### DISCUSSION TITLE
compile hex using make

### DISCUSSION THREAD

**Discussion by macleayislandbruce** [iNavFlight/inav#7508] [Q&A] [answered]
Hi, I have just successfully compiled a hex file using make as described in several places. The resulting hex file however is inav_4.0.0_OMNIBUSF7.hex ! Naturally configurator 3.0.1 won't talk to v4.0.0 only cli works. From cli it appears the hex file is good. 

So first question how to set compile options for version 3.0.2? 

Second question, the compilation process included a lot of files which I don't want included, eg for hardware I don't have or that the board cannot address. Where can I edit/remove hardware options? I have looked at lots of make files, most say "DO NOT EDIT this file" 

My build invironment is debian9 linux, I had to downgrade cmake from 3.8 to 3.13 to use the inav ruleset.
Cheers Bruce

**Comment by macleayislandbruce** [comment]
Update, after reading another post I found that editing the version number in CMakeLists.txt solved question 1, Why is it currently V4.0.0, who knows but changing it back works and Configurator works with it ok.

**Comment by DzikuVx** [comment]
Because current version will be called 4.0 not 3.0. You need latest builds of Configurator if you want to build from source http://seyrsnys.myzen.co.uk/inav-configurator-next/

**Reply by macleayislandbruce** [reply]
Thanks Pawel, I will have a go at building a new Configurator. Also a well done Pawel! I have built a LoRa radio system using STM32F103 (blue pill) and eByte sx1262 at each end. The drone end then uses your code to generate S-Bus in a simple way, works a treat. I am a self taught programmer and your work saved me quite a bit of brain strain. 

I am still looking for an answer to question 2. While I have some competency in C++, I haven't used cmake/make before.

Cheers Bruce

**Reply by macleayislandbruce** [reply]
re: question 2
Is this just a matter of editing target.c and removing unrequired hardware from BUSDEV, therefor make won't link in libraries that are not required?
Now have working V4.0.0 Configurator and hex which all work together, many thanks for the advice

**Comment by DzikuVx** [comment, marked as answer]
makefile is generated by `cmake` The best way is to undefined certain functions like `USE_***` and linker will not include not used files

**Reply by macleayislandbruce** [reply]
Thanks Pawel. I have now modified source code to remap pins and added LIS3MDL magnetometer, now have working mag so I am a lot more comfortable with the build process. Thanks for advice on undefining functions that makes sense.
